rcl/include/rcl/macros.h

```diff
@@ -15,18 +15,11 @@
 #ifndef RCL__MACROS_H_
 #define RCL__MACROS_H_
 
-#ifdef __cplusplus
-extern "C"
-{
-#endif
```

Comment:
Why should this be removed?

Reply:
Because there is nothing to specify the linkage of. And were there any, we’d want to use whatever linkage is specified by `rcutils`

Reply:
Almost all C headers in the `rcl` packages do use `extern "C"` - no matter if they do contain things which require it or not. The rational is that since the package is providing a C-only API. Always stating it explicitly ensures that it is being used - even if additional code is added to a header later - since it might otherwise be overlooked.

Therefore a :-1: from me to remove it. I would rather propose to add it to the headers where it is currently not.

Reply:
> since it might otherwise be overlooked.

i believe this is reasonable.

Reply:
I think it's needless clutter and misleading to keep it in. If a macro in this `macros.h` file declared a function, that enclosing `extern "c"` would falsely suggest that the function uses C linkage.

Reply:
I don't feel strongly about it, but leaving seems harmless to me.

Anyways, removing it seems orthogonal to this PR and could be proposed separately.

Reply:
I also don't feel super strongly about it, but on balance I would rather leave the `extern "C"` in.  It just keeps all of the headers consistent, and if we did ever introduce a C-linkage symbol in it, it would prevent it from being mangled.

Reply:
@rotu With the majority of feedback leaning toward keeping the `extern "C"` blocks please revert this part of the patch so we can merge the deduplication.

Reply:
I reverted the removal of `extern C` in b8f11aa0ef7672fed01d298d5915fbb619ccf341.